lib/bn.js

```diff
@@ -476,8 +476,14 @@
   };
 
   BN.prototype.toNumber = function toNumber () {
-    assert(this.bitLength() <= 53, 'Number can only safely store up to 53 bits');
-    return parseInt(this.toString(), 10);
+    var length = this.bitLength();
+    assert(length <= 53, 'Number can only safely store up to 53 bits');
+    if (length <= 26) {
+      var ret = this.words[0] & 0x3ffffff;
+      return (this.negative !== 0) ? (ret * -1) : ret;
+    } else {
+      return parseInt(this.toString(), 10);
```

Comment:
I was thinking about that, but that would mean just limiting toNumber to 52 bits instead of 53 and be done with it.